src/renderers/dom/shared/__tests__/ReactDOMAttribute-test.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @emails react-core
+ */
+
+'use strict';
+
+describe('ReactDOM unknown attribute', () => {
+  var React;
+  var ReactDOM;
+
+  beforeEach(() => {
+    jest.resetModules();
+    React = require('react');
+    ReactDOM = require('react-dom');
+  });
+
+  function testUnknownAttributeRemoval(givenValue) {
+    var el = document.createElement('div');
+    ReactDOM.render(<div unknown="something" />, el);
+    expect(el.firstChild.getAttribute('unknown')).toBe('something');
+    ReactDOM.render(<div unknown={givenValue} />, el);
+    expect(el.firstChild.hasAttribute('unknown')).toBe(false);
+  }
+
+  function testUnknownAttributeAssignment(givenValue, expectedDOMValue) {
+    var el = document.createElement('div');
+    ReactDOM.render(<div unknown="something" />, el);
+    expect(el.firstChild.getAttribute('unknown')).toBe('something');
+    ReactDOM.render(<div unknown={givenValue} />, el);
+    expect(el.firstChild.getAttribute('unknown')).toBe(expectedDOMValue);
+  }
+
+  describe('unknown attributes', () => {
+    it('removes values null and undefined', () => {
+      testUnknownAttributeRemoval(null);
+      testUnknownAttributeRemoval(undefined);
+    });
+
+    it('removes unknown attributes that were rendered but are now missing', () => {
+      var el = document.createElement('div');
+      ReactDOM.render(<div unknown="something" />, el);
+      expect(el.firstChild.getAttribute('unknown')).toBe('something');
+      ReactDOM.render(<div />, el);
+      expect(el.firstChild.hasAttribute('unknown')).toBe(false);
+    });
+
+    it('passes through strings', () => {
+      testUnknownAttributeAssignment('a string', 'a string');
+    });
+
+    it('coerces numbers and booleans to strings', () => {
+      testUnknownAttributeAssignment(0, '0');
+      testUnknownAttributeAssignment(-1, '-1');
+      testUnknownAttributeAssignment(42, '42');
+      testUnknownAttributeAssignment(9000.99, '9000.99');
+      testUnknownAttributeAssignment(true, 'true');
```

Comment:
That was our conclusion, recorded in our notes, and I still question which is the right way. 

I think our logic was that if we filter out booleans for unknown props, then we would also consider changing that behavior for known attributes, and that would create more work to get this landed as well as more API changes for users.

It's true that, long term, we should probably stop stringifying booleans and filter them out.